.github/workflows/test.yml

```diff
@@ -119,10 +119,31 @@ jobs:
       - name: Configure Windows (R-devel)
         if: startsWith(runner.os, 'Windows') && matrix.config.r == 'devel'
         run: |
+          $rtools42_home = "$(Rscript.exe -e "cat(Sys.getenv('RTOOLS42_HOME'))")"
+
           # c.f. https://github.com/wch/r-source/blob/f1501504df8df1668a57d3a1b6f80167f24441d3/src/library/profile/Rprofile.windows#L70-L71
-          echo "C:\rtools42\x86_64-w64-mingw32.static.posix\bin"           | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append ;
-          echo "C:\rtools42\usr\bin"                                       | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append ;
+          echo "${rtools42_home}\x86_64-w64-mingw32.static.posix\bin"      | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append ;
+          echo "${rtools42_home}\usr\bin"                                  | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append ;
           echo "$(Rscript.exe -e 'cat(normalizePath(R.home()))')\bin\x64"  | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append ;
+
+          # The default linker of rustc is "x86_64-w64-mingw32-gcc", however, there's no such name of executable in Rtools42
+          New-Item -Path .cargo -ItemType Directory -Force
+          @'
+          [target.x86_64-pc-windows-gnu]
+          linker = "x86_64-w64-mingw32.static.posix-gcc.exe"
+          '@ | Out-File -FilePath .cargo/config.toml -Encoding utf8 -Append ;
+
+          # rustc adds -lgcc_eh flag to the compiler, but there's no libgcc_eh in Rtools42 (this might be a bug in MSYS2 itself?, but Rtools40
+          # has a workaround for this).
+          cp `
+            "${rtools42_home}\x86_64-w64-mingw32.static.posix\lib\gcc\x86_64-w64-mingw32.static.posix\10.3.0\libgcc.a" `
+            "${rtools42_home}\x86_64-w64-mingw32.static.posix\lib\gcc\x86_64-w64-mingw32.static.posix\10.3.0\libgcc_eh.a"
```

Comment:
This is wrong.
There are 3 GCC libraries:
- `libgcc.a`
- `libgcc_eh.a`
- `libgcc_s.a`

`libgcc.a` contains compiler builtins among the other things but has nothing to do with unwinding.

`libgcc_eh.a` is static unwind library, while `libgcc_s.a` is import library for shared unwind library (on 64-bit it's `libgcc_s_seh-1.dll`).

Reply:
Does this help?

> ```
> # 'gcc' when build with --disable-shared does not produce a standalone 'libgcc_eh.a'.
> # Instead, everything is put into 'libgcc.a'.
> # To please compilers that hardcode -lgcc_eh, such as 'rustc', 'libgcc.a' is duplicated as 'libgcc_eh.a'.
> ```
>(https://github.com/r-windows/rtools-packages/blob/2407b23f1e0925bbb20a4162c963600105236318/mingw-w64-gcc/PKGBUILD#L313-L316)

We confirmed `libgcc.a` and `libgcc_eh.a` are the same binaries. We don't figure out what `libgcc_s` is yet, so thanks for the details.

Reply:
Oh, that's really unfortunate from GCC to that.